include/shark/Algorithms/DirectSearch/Indicators/LeastContributorApproximator.h

```diff
@@ -562,40 +330,196 @@ struct LeastContributorApproximator {
+	template<typename Set>
+	bool isRandomPointDominated( Set const& s, typename Set::iterator point )
+	{
+		point->m_sample = point->m_point;
+		for( unsigned int i = 0; i < point->m_sample.size(); i++ ) {
+			point->m_sample[ i ] += Rng::uni( 0., 1. ) * ( point->m_boundingBox[ i ] - point->m_point[ i ] );
+		}
+
+		bool successful = true;
+		for( unsigned int i = 0; i < point->m_influencingPoints.size(); i++ ) {
+			point->m_noOperations += point->m_sample.size() + 1;
+			bool inside = true;
+			for( unsigned int j = 0; j < point->m_sample.size(); j++ ) {
+				if( point->m_influencingPoints[i]->m_point[ j ] > point->m_sample[ j ] ) {
+					inside = false;
+					break;
+				}
+			}
+
+			if( inside ) {
+				successful = false;
+				break;
+			}
+		}
+
+		return successful;
+	}
+
+	template<typename iterator>
+	double deltaForPoint( iterator point, unsigned int R )
+	{
+
+		return( ::sqrt( 0.5 * ((1. + m_gamma) * ::log( static_cast<double>( R ) ) + m_logFactor ) / point->m_noSamples ) * point->m_boundingBoxVolume );
+
+	}
+
+	/// \brief Computes bounding boxes and their volume for the range of points defined by the iterators.
+	///
+	/// \param [in] begin Iterator pointing to the first valid point.
+	/// \param [in] end Iterator pointing right after the last valid point.
+	template<class SetIterator>
+	void computeBoundingBoxes(SetIterator begin, SetIterator end )
+	{
+		for(auto it = begin; it != end; ++it ) {
+			auto& p1 = *it;
+			for(auto itt = begin; itt != end; ++itt ) {
+
+				if( itt == it )
+					continue;
+
+				auto& p2 = *itt;
+
+				unsigned int coordCounter = 0;
+				unsigned int coord = 0;
+				for( unsigned int o = 0; o < p1.m_point.size(); o++ ) {
+					if( p2.m_point[ o ] > p1.m_point[ o ] ) {
+						coordCounter++;
+						if( coordCounter == 2 )
+							break;
+						coord = o;
+					}
+				}
+
+				if( coordCounter == 1 && p1.m_boundingBox[ coord ] > p2.m_point[ coord ] )
+					p1.m_boundingBox[ coord ] = p2.m_point[ coord ];
+			}
+
+			it->m_boundingBoxVolume = 1.;
+			for(unsigned int i = 0 ; i < it->m_boundingBox.size(); i++ ) {
+
+				it->m_boundingBoxVolume *= it->m_boundingBox[ i ] - it->m_point[ i ];
+
+			}
+
+			for(auto itt = begin; itt != end; ++itt ) {
+				if( itt == it )
+					continue;
+
+				bool isInfluencing = true;
+				double vol = 1.;
+				for( unsigned int i = 0; i < it->m_point.size(); i++ ) {
+
+					if( itt->m_point[ i ] >= it->m_boundingBox[ i ] ) {
+						isInfluencing = false;
+						break;
+					}
+					vol *= std::max( itt->m_point[ i ], it->m_point[ i ] ) - it->m_boundingBox[ i ];
+				}
+				if( isInfluencing ) {
+					itt->m_overlappingVolume = vol;
+					it->m_influencingPoints.push_back( itt );
+				}
+			}
+			std::sort( 
+				it->m_influencingPoints.begin(),
+				it->m_influencingPoints.end(),
+				[]( SetIterator i1, SetIterator i2 ){
+					return i1->m_overlappingVolume > i2->m_overlappingVolume;
+				}
+			);
+		}
+	}
 };
```

Comment:
I did not check the large block above this point because you plan to look at this anyway soon.